tests/test_multichannel.py

```diff
@@ -902,3 +903,90 @@ def test_click_multi():
     assert np.allclose(yout[..., :100], click)
     assert np.allclose(yout[..., 1000:1100], click)
     assert np.allclose(yout[..., 2000:2100], click)
+
+
+def test_nnls_multi(s_multi):
+
+    # Verify that a stereo melspectrogram can be reconstructed
+    # for each channel individually
+    S, sr = s_multi
+
+    # multichannel  
+    mel_basis = librosa.filters.mel(sr, n_fft=2*S.shape[-2]-1)
+    M = np.einsum('...ft,mf->...mt', S, mel_basis)
+    S_recover = librosa.util.nnls(mel_basis, M)
+
+    # channel 0
+    M0 = np.einsum('...ft,mf->...mt', S[0], mel_basis)
+    S0_recover = librosa.util.nnls(mel_basis, M0)
+
+    # channel 1
+    M1 = np.einsum('...ft,mf->...mt', S[1], mel_basis)
+    S1_recover = librosa.util.nnls(mel_basis, M1)
+
+    # Check each channel
+    assert np.allclose(S_recover[0], S0_recover)
+    assert np.allclose(S_recover[1], S1_recover)
+
+    # Check that they're not both the same
+    assert not np.allclose(S0_recover, S1_recover)
+
+
+# -- feature inversion tests
+@pytest.mark.parametrize("power", [1, 2])
+@pytest.mark.parametrize("n_fft", [1024, 2048])
+def test_mel_to_stft_multi(power,  n_fft):
+    srand()
+
+    # Make a random mel spectrum, 4 frames
+    mel_basis = librosa.filters.mel(22050, n_fft, n_mels=128)
+
+    stft_orig = np.random.randn(2, n_fft // 2 + 1, 4) ** power
```

Comment:
This seems like it shouldn't work when power=1; randn can produce negative values, but mel_to_stft can't.  It's not a huge deal here, but it might put us in a weird part of the solution space that could cause trouble.

Reply:
I just adapted the `mel_to_stft` test in `test_features.py`. We might want to change that too then?

Reply:
Nah, we can leave it as is for consistency / minimal surprise.  I was just flagging it as a possible thing to take note of.